### PR TITLE
vignette addition and dependency help update

### PR DESCRIPTION
**Pull request by diegobilski**
Adds vignette on updating scripts to rsyncrosim v2. 
Changes ` rsyncrosim` to rsyncrosim on vignette titles. 
Change on dependency function help.

### CODE REVIEW

File: R/dependency.R [hunk 1]
```diff
@@ -17,19 +17,13 @@ NULL
 #' Otherwise, returns list of saved or error messages for each dependency of each 
 #' scenario.
 #'
-#' Note that the order of dependencies can be important - dependencies added most
```
Comment (katieb1) [inline R/dependency.R:20]: I think you might have overwritten the updated docs for the dependencies. I would revert your changes to this script and then just make the one tweak to the priority wording and re-commit.

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+```{r deprecated, eval=FALSE}
+addon()
+disableAddon()
+enableAddon()
+```
+
+## Library
+
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Add code chunk with output to show update to `info()` function. Output could be added in a "hacky" way for clarity.

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+In SyncroSim 3, the concept of addon packages no longer exists. All addon packages have either been converted to standalone packages (e.g., `burnP3PlusPrometheus`) and can be added to a library without having to load a base package first, or incorporated directly into the base packages (e.g., `stsimsf` is now part of `stsim`). Therefore, the following functions have been removed:
+
+```{r deprecated, eval=FALSE}
+addon()
+disableAddon()
+enableAddon()
+```
+
+## Library
+
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:91]: Add overview sentence about changes to library functions

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
+
+The `package` argument in the `ssimLibrary(package)` function has been renamed to `packages`
+
+```{r library, eval=FALSE}
+myLibrary <- ssimLibrary(packages = c())
+```
+
+The `addon` and `template` arguments have also been removed.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I would add this to above because it is unclear it relates to the `ssimLibrary()` function

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+addon()
+disableAddon()
+enableAddon()
+```
+
+## Library
+
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
+
+The `package` argument in the `ssimLibrary(package)` function has been renamed to `packages`
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: consistency in punctuation

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+```{r library, eval=FALSE}
+myLibrary <- ssimLibrary(packages = c())
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I would show the old way of doing things versus the new way for comparison

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+The `addon` and `template` arguments have also been removed.
+
+## Dependencies
+
+To view the existing dependencies for a scenario, use:
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Add overview sentence on changes to dependencies. Add old way of doing things versus new way to highlight differences.

Comment (katieb1) [reply vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Also I would generate some output to show what happens when you add dependencies

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+dependency(myScenario) <- c("Scenario 1", "Scenario 2")
+```
+
+Note that this will remove any previously set dependencies.
+Also, the order in which the dependencies are listed can be important. In this example, `Scenario 1` takes precedence over `Scenario 2`.
+
+
+## Multiprocessing
+
+The `run()` function no longer has the `jobs` argument. Use the `core_Multiprocessing` datasheet to set the number of cores to use. Note that this will affect every scenario run.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I think this part needs more detail. For instance, it's unclear that the `core_Multiprocessing` datasheet is set at the library scope. Also include an example of how to set the `core_Multiprocessing` datasheet.

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+
+## Variable naming
+
+The **ID** in variable names have been changed to **Id**. It is generally safe to substitute `ID` for `Id` throughout your script, unless that affects any variables not related to SyncroSim.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I think this also needs more detail. It's specifically the "ID" in the primary key that has been updated. Some datasheets might have additional columns that have "ID" in them that the package developer has set. Give example

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+## Core datasheets
+
+The `corestime_` datasheets are now simply called `core_`.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Add link to "System Datasheets" reference guide on syncrosim docs. More details on what the `core_` means (give a brief summary based on content in the System Datasheets reference guide overview).

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+## stsimsf
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Remove section on stsimsf - this should go in the ST-Sim docs instead

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+
+The ST-Sim Stock & Flow package, `stsimsf` have been incorporated into ST-Sim. Substitute all `stsimsf_` datasheets throghout your script for `stsim_`.
+The only exception is the `stsimsf_OutputOptions` that is now called `stsim_OutputOptionsStockFlow`.
+
+## Other removed functions
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Move up above with other functions that have been removed (I created a section called "Deprecated functions").

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+addBreakpoint()
+breakpoint()
+deleteBreakpoint()
+
+ssimUpdate()
+updatePackage()
+```
+
+## Other NEW functions
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:206]: Move up above as well and give each new function it's own subsection. Write a brief description about each of these new functions.

**Review by katieb1 — CHANGES_REQUESTED** [review]
Thanks, Diego! Looks like all the content is there, I think we just need to expand a bit to add more details on what is being updated and add more examples and comparisons with the old code. Could be nice to have an example library that you run through the vignette to show the output of the updated/new functions. 

I think also the dependencies docstrings need to be fixed! See my comment

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 14]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
 
-In SyncroSim 3, the concept of addon packages no longer exists. All addon packages have either been converted to standalone packages (e.g., `burnP3PlusPrometheus`) and can be added to a library without having to load a base package first, or incorporated directly into the base packages (e.g., `stsimsf` is now part of `stsim`). Therefore, the following functions have been removed:
+In SyncroSim 3, the concept of addon packages no longer exists. All addon packages have either been converted to standalone packages (e.g., `burnP3PlusPrometheus`), and can be added to a library without having to load a base package first, or incorporated directly into the base packages (e.g., `stsimsf` is now part of `stsim`). Therefore, the following functions have been removed:
 
 ```{r deprecated, eval=FALSE}
 addon()
 disableAddon()
 enableAddon()
+
+datasheetRaster() # previously deprecated function. Use datasheetSpatRaster() instead.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: This seems a bit unrelated to the above text to include these other functions in the same code chunk. I think each group of deprecated functions deserves a little bit of text explaining why they are no longer needed.

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 15]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
+
+# use
+myLibrary <- ssimLibrary(packages = c("stsim", "helloworldSpatial"))
+```
+```{r load session run, warning = FALSE, echo = FALSE}
+# results for the section above
+library(rsyncrosim)
+mySession <- session()
+myLibrary <- ssimLibrary("Changes to rsyncrosim v2.ssim", packages=c("stsim", "helloworldSpatial"))
+scenario1 <- scenario(myLibrary, scenario = "Scenario 1")
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I'm not sure why this other code is in here that doesn't have to do with the `ssimLibrary()` changes?

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 16]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
+ssimUpdate()
+updatePackage()
 ```
 
 ## Library
 
-`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
-
-The `package` argument in the `ssimLibrary(package)` function has been renamed to `packages`
+Given this added flexibility, the `addon` argument has been removed from the `ssimLibrary()` function, and the `package` argument has been renamed to `packages`.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Maybe something more like:

> Now that addon packages no longer exist and SyncroSim libraries can support multiple packages, the following changes have been made to the `ssimLibrary()` function:
> * The `addon` argument has been removed.
> * The `template` argument has been removed. This is a temporary change, but in the meantime you can download template libraries from<a href="https://cloud.syncrosim.com/" target="_blank">SyncroSim Cloud</a>.
> * The `package` argument has been renamed to `packages` (plural).

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 17]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
+scenario2 <- scenario(myLibrary, scenario = "Scenario 2")
+scenario3 <- scenario(myLibrary, scenario = "Scenario 3")
+myScenario <- scenario(myLibrary, "my Scenario")
+dependency(myScenario) <- "Scenario 1"
 ```
 
-The `addon` and `template` arguments have also been removed.
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*. See the <a href="https://syncrosim.github.io/rsyncrosim/articles/a01_rsyncrosim_vignette_basic.html#access-model-metadata" taret="_blank">Introduction to rsyncrosim</a> for details.
+
+The `template` and `addon` arguments have also been removed. You can download template libraries from <a href="https://cloud.syncrosim.com/" target="_blank">SyncroSim Cloud</a>.
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I would remove this line - it is referring to changes to `ssimLibrary()` but you bring up changes to `info(ssimLibrary)` in between so it is confusing

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd (hunk 7 above)
Comment (katieb1) [reply vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Maybe something like:

> Scenario dependencies are now set using the `<-` operator in rsyncrosim. See below for examples on how the usage of the `dependency()` function has changed.

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 18]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
 
 ## Dependencies
 
 To view the existing dependencies for a scenario, use:
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: Make structure of this sentence consistent with below

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 19]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
 dependency(myScenario)
 ```
 
-To set dependencies for a scenario, use:
+To set dependencies for a scenario:
 
 ```{r set dependencies, eval=FALSE}
-dependency(myScenario) <- c("Scenario 1", "Scenario 2")
-```
+# Change from:
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I would make the comments consistent across sections when comparing old code to new code. So, when demonstrating old code, could be:

> `# rsyncrosim version 1:`

and new code could be:

> `# rsyncrosim version 2:`

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd [hunk 20]
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
...
 
-The **ID** in variable names have been changed to **Id**. It is generally safe to substitute `ID` for `Id` throughout your script, unless that affects any variables not related to SyncroSim.
+The `run()` function no longer has the `jobs` argument. Use the `core_Multiprocessing` library datasheet to set the number of cores to use. Note that this will affect every scenario run.
 
-## Core datasheets
+```{r multiprocessing, eval=FALSE}
+multiprocessing <-
+    data.frame(EnableMultiprocessing = TRUE,
+               MaximumJobs = 6,
+               EnableMultiScenario = TRUE) # optional, to run multiple scenarios in parallel
```
Comment (katieb1) [inline vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: I would remove this argument - users couldn't set this from the `run()` function before anyway

**Review by katieb1 — CHANGES_REQUESTED** [review]

File: vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd (hunk 15 above)
Comment (diegobilski) [reply vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd]: this is just to show the addition of two packages, plus creating a couple of scenarios to be used later
